src/main/java/gregtech/common/pipelike/itempipe/net/ItemNetWalker.java

```diff
@@ -38,32 +47,71 @@ protected ItemNetWalker(World world, BlockPos sourcePipe, int distance, List<Ite
     }
 
     @Override
-    protected PipeNetWalker createSubWalker(World world, BlockPos nextPos, int walkedBlocks) {
+    protected PipeNetWalker createSubWalker(World world, EnumFacing facingToNextPos, BlockPos nextPos, int walkedBlocks) {
         ItemNetWalker walker = new ItemNetWalker(world, nextPos, walkedBlocks, inventories, minProperties);
         walker.facingToHandler = facingToHandler;
         walker.sourcePipe = sourcePipe;
+        walker.filters.addAll(filters);
+        List<Predicate<ItemStack>> moreFilters = nextFilters.get(facingToNextPos);
+        if (moreFilters != null && !moreFilters.isEmpty()) {
+            walker.filters.addAll(moreFilters);
+        }
         return walker;
     }
 
     @Override
     protected void checkPipe(IPipeTile<?, ?> pipeTile, BlockPos pos) {
+        for (List<Predicate<ItemStack>> filters : nextFilters.values()) {
+            if (!filters.isEmpty()) {
+                this.filters.addAll(filters);
+            }
+        }
```

Comment:
The pipe net is not my specialty, so this may be an obviously wrong question, but is it correct to check `nextFilters` here?

As far as I can see, `nextFilters` is only updated in `isValidPipe`, which is called after `checkPipe` in `PipeNetWalker#checkPos`, so I was wondering if checking `nextFilters` here would provide up to date information on any filters.

Reply:
Honestly, i have no fucking idea. It was over a month ago...

Reply:
`nextFilters` acts as a queue. So in `isValidPipe` all filters an all sides are added, but the filters should only get added if the walker actually walked in that direction, so in the next `checkPipe` it gets all filters from before and adds them (idk why i iterate over the map).